Junk4Dollars-app/Auctions/AuctionListViewController.swift

```diff
@@ -43,6 +45,12 @@ public class AuctionListViewController: UIViewController, AuctionListView {
         }
     }
 
+    public func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) -> Void {
+        self.auctionsDataSource.auctionFor(indexPath: indexPath) { id in
+            print("ROW => \(indexPath.row) ID =>", id)
+        }
```

Comment:
Here is where you're using the callback.

`self.auctionsDataSource.auctionFor` returns an `Auction`, so why don't you use that return value, instead of giving it a callback?
```
auction = self.auctionsDataSource.auctionFor(indexPath: indexPath)
print("ROW => \(indexPath.row) ID =>", auction.id)
```